src/change_stream.ts

```diff
@@ -365,13 +372,24 @@ export class ChangeStream<TSchema extends Document> extends TypedEventEmitter<Ch
   tryNext(): Promise<Document | null>;
   tryNext(callback: Callback<Document | null>): void;
   tryNext(callback?: Callback<Document | null>): Promise<Document | null> | void {
+    this._setIsIterator();
     return maybePromise(callback, cb => {
       getCursor(this, (err, cursor) => {
         if (err || !cursor) return cb(err); // failed to resume, raise an error
         return cursor.tryNext(cb);
       });
     });
   }
+
+  _setIsEmitter(): void {
```

Comment:
Can you make these helpers taking a ChangeStream like the `errorIfIsX` functions, and inline those since they are only used once? That keeps the implementation details a little more hidden and the class API more streamlined.